src/entry_completion.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2019, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use glib::Cast;
+use glib::object::IsA;
+use glib::translate::*;
+use gtk_sys;
+use Entry;
+use EntryCompletion;
+use Widget;
+
+pub trait EntryCompletionExtManual: 'static {
+    fn get_entry(&self) -> Option<Entry>;
+}
+
+impl<O: IsA<EntryCompletion>> EntryCompletionExtManual for O {
+    fn get_entry(&self) -> Option<Entry> {
+        let widget: Option<Widget> = unsafe {
+            from_glib_none(gtk_sys::gtk_entry_completion_get_entry(self.as_ref().to_glib_none().0))
+        };
+        match widget {
+            Some(w) => w.downcast().ok(),
```

Comment:
This should probably panic. If it's not an entry then things are broken.

Reply:
I don't like panicking in library calls. :-/

Reply:
`from_glib_none()` also panics if the thing there is of the wrong type. A panic is the only sensible thing to do here if basic assumptions are wrong, it's nothing a user can handle in a meaningful way.

Reply:
Yes but in here, it comes from us in a "normal" way supposed to be fully working.

Reply:
I don't understand what you mean. If the widget there is not an entry, then there's a bug in GTK somewhere and the basic assumptions of this API are broken. That's nothing the user can handle in a meaningful way.

Reply:
Isn't it exactly why we have the `Result` type in rust?

Reply:
Not from my point of view but I'll replace with a call to `expect`.